models/DataObject/ClassDefinition/Data/Numeric.php

```diff
@@ -449,6 +449,17 @@ public function getFilterConditionExt(mixed $value, string $operator, array $par
             return $key . ' ' . $operator . ' ' . $value . ' ';
         }
 
+        if ($operator === 'in') {
+            $formattedValues = implode(
+                ',',
+                array_map(function ($value) use ($db) {
+                    return (float)$value;
+                }, explode(',', $value))
+            );
```

Comment:
Can be replaced by this:
```suggestion
            $formattedValues = implode(',', array_map(floatval(...), explode(',', $value)));
```